### PR TITLE
Allow for adding / removing properties from DefaultConfig

### PR DESCRIPTION
**Pull request by bdemers**
Currently `put`, `putAll`, `remove`, and `clear` cannot be used as they are backed by an unmodifiable map.

If `props` is truly not to be modified, we should instead throw an exception directly from the corresponding method (to make it more obvious)

### CODE REVIEW

**Comment by mraible** [comment]
LGTM 👍 @lhazlewood @mrioan @dogeared What do you think?

File: extensions/servlet/src/test/groovy/com/stormpath/sdk/servlet/config/impl/DefaultConfigFactoryTest.groovy [hunk 1]
```diff
@@ -142,6 +143,73 @@ class DefaultConfigFactoryTest {
         config = new ConfigLoader().createConfig(new MockServletContext())
     }
 
+    @Test
+    public void putProperty() {
```
Comment (mrioan) [inline extensions/servlet/src/test/groovy/com/stormpath/sdk/servlet/config/impl/DefaultConfigFactoryTest.groovy]: No show stopper but the name of the test should a) start with the word `test` and b) help understand what it is testing; like `testPutPropertyIsAbleToModifyTheMap()`. Just for future reference

**Comment by mrioan** [comment]
Just a simple educational comment from, no need to really change the code this time. LGTM. Thanks @bdemers

**Comment by bdemers** [comment]
@mrioan easy enough to fix.

Do you guys usually squash and rebase before merging PR's?

**Comment by lhazlewood** [comment]
I prefer a squash, but we don't usually do that.

**Comment by bdemers** [comment]
updated @since tags to next release and squashed

**Comment by mrioan** [comment]
@bdemers Feel free to merge. Thanks

**Pull request merged**